src/message.rs

```diff
@@ -1236,7 +1236,7 @@ pub async fn delete_msgs(context: &Context, msg_ids: &[MsgId]) -> Result<()> {
     for msg_id in msg_ids.iter() {
         let msg = Message::load_from_db(context, *msg_id).await?;
         if msg.location_id > 0 {
-            delete_poi_location(context, msg.location_id).await;
+            delete_poi_location(context, msg.location_id).await?;
```

Comment:
Just a note: Now the deletion will be cancelled on SQL errors. That's OK though because if the other SQL statements here error, it will also be cancelled.